rest_framework/compat.py

```diff
@@ -6,12 +6,32 @@
 # flake8: noqa
 from __future__ import unicode_literals
 
+import warnings
+
 import django
 from django.conf import settings
 from django.db import connection, transaction
 from django.utils import six
 from django.views.generic import View
 
+from rest_framework import VERSION
+
+
+def deprecated(since, message):
+    current_version = [int(i) for i in VERSION.split('.')]
+
+    assert current_version[0] == since[0], "Deprecated code must be removed before major version change. Current: {0} vs Deprecated Since: {1}".format(current_version[0], since[0])
```

Comment:
Might have been confused by the `since` argument. Maybe it should be named `target_version` or something similar.